spec/models/submit_spec.rb

```diff
@@ -114,5 +114,32 @@ def add_participant(contest_owner = owner)
       expect(contest_solutions[0][:solutions][0][:attempts]).to eq(1)
       expect(contest_solutions[1][:solutions][0][:attempts]).to eq(1)
     end
+
+    it 'sorts problems in order: required, created_at' do
+      later_optional_problem = create(:contest_problem, contest: participation.contest, required: false,
```

Comment:
można stworzyć pomocniczą funkcję do setupu problemów, bo kod jest dosyć powtarzalny

Reply:
Mam mieszane uczucia – napisałem komentarz na 10 linii, po czym skasowałem, bo nie jest warte dyskusji. Miałem pomysł na bardziej uniwersalne rozwiązanie, ale kosztem czytelności. Ten kod jest powtarzalny, ale nie chciałbym go skracać, bo jest ultraprosty – dodanie naszych wywołań to skomplikuje.